lib/transforms/subsetFonts.js

```diff
@@ -567,15 +567,21 @@ module.exports = function (options) {
                                 var fontAsset = assetGraph.findAssets({ url: fontUsage.props.src })[0];
 
                                 // Always preload unsubsetted font files, they might be any format, so can't be clever here
-                                return new AssetGraph.HtmlPreloadLink({
-                                    hrefType: 'rootRelative',
-                                    to: fontAsset,
-                                    as: 'font'
-                                });
-                            });
 
-                            preloadRelations.forEach(function (rel) {
-                                rel.attach(htmlAsset, 'before', insertionPoint);
+                                var preloadHint = assetGraph.findRelations({
+                                    type: 'HtmlPreloadLink',
+                                    as: 'font',
+                                    to: fontAsset
```

Comment:
This will find any preload relation to the font, including ones on different pages. We need to match the from Html asset as well